### PR TITLE
[docs] Add privacy access feature Explainer

### PR DESCRIPTION
**Pull request by lparisc**
## Change Description

Resolves devrel-1350 - Private access: Create Explainer doc in nodeos/Concepts

## Change Type
**Select *ONE*:**
- [x] Documentation

- [ ] Stability bug fix

- [ ] Other

- [ ] Other - special case

## Testing Changes
**Select *ANY* that apply:**
- [ ] New Tests

- [ ] Existing Tests

- [ ] Test Framework

- [ ] CI System

- [ ] Other

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

**Review by dimas1185 — APPROVED** [review]

File: docs/01_nodeos/07_concepts/15_private_chain_access/index.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+[[info | Access controls via consensus]]
+| Along with the inherent features of the EOSIO blockchain, the *Private Access Control* feature achieves security, robustness, and coordination through EOSIO consensus. The protocol feature can be enabled and participants added or removed from a *security group* only after the block that contains the corresponding transaction is approved by a supermajority of producer participants and the block becomes irreversible.
+
+## Requirements
+
+The *Private Access Control* feature must meet the following requirements:
+- The SECURITY_GROUP protocol feature must be activated first via EOSIO consensus.
+- Once activated, only nodes with valid signed credentials are allowed to connect.
+- valid nodes must establish TLS connections to communicate securely with other peers.
```
Comment (bobgt) [inline docs/01_nodeos/07_concepts/15_private_chain_access/index.md:22]: valid -> Valid

File: docs/01_nodeos/07_concepts/15_private_chain_access/index.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
+---
+content_title: Private Chain Access Control
+link_text: Private Chain Access Control
+---
+
+## Overview
+
+The *Private Access Control* or *Security Group* feature provides a two-layer security solution for blockchain administrators to control which participants can access and use a private EOSIO network. The first security layer enforces TLS connections in the peer-to-peer network protocol; only those participants with a properly signed certificate are able to establish a TLS connection and communicate securely with other peers. The second security layer, which is optional, brings the concept of a *privacy group* or *security group*; if enabled, only those participants in the group are allowed to access data from the private network. Therefore, the first layer controls “connection” access while the second layer manages “data” access. Participants are considered logical entities with a designated EOSIO name and may host multiple nodes, potentially deployed in [Block Vault](../../03_plugins/blockvault_client_plugin/index.md) mode.
```
Comment (bobgt) [inline docs/01_nodeos/07_concepts/15_private_chain_access/index.md:8]: Is the full name of the feature *Private Access Control* or *Private Chain Access Control*? If the latter, then we have to change it in all instances. If the former, then no action required!

File: docs/01_nodeos/07_concepts/15_private_chain_access/index.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+## Concept
+
+Most businesses need to keep data private for regulatory compliance, customer privacy, or both. Since blockchain solutions have a history of public data and open architecture, businesses need certain privacy and security guarantees before they can adopt blockchain technology. The *Private Access Control* feature aims to bring EOSIO closer to the Enterprise by providing a general business security solution that can be used across a variety of [Use Cases](#use-cases).
+
+[[info | Access controls via consensus]]
+| Along with the inherent features of the EOSIO blockchain, the *Private Access Control* feature achieves security, robustness, and coordination through EOSIO consensus. The protocol feature can be enabled and participants added or removed from a *security group* only after the block that contains the corresponding transaction is approved by a supermajority of producer participants and the block becomes irreversible.
```
Comment (bobgt) [inline docs/01_nodeos/07_concepts/15_private_chain_access/index.md:15]: Maybe link the "irreversible" text to the glossary? 
https://developers.eos.io/welcome/latest/glossary/index/#irreversible-block

File: docs/01_nodeos/07_concepts/15_private_chain_access/index.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+- Once connected, all peer-to-peer communications must be encrypted - TLS satisfies this.
+- Optionally, a *security group* can be used to manage participant data access to the network:
+  * If at least one participant is added to the group, data access control is enabled.
+  * If no participants are added to the group, data access control is disabled.
+  * When data access control is disabled, all participants in the group are allowed data access.
+  * When data access control is enabled, only those participants in the group are allowed data access.
+
+The secure connection requirements are met by establishing TLS connections among the participant nodes. This is considered sufficient from a security standpoint since the peer-to-peer communications are encrypted. The optional data access control requirements are met through the *security group* which adds further data access control on a per-participant basis. Adding and removing participans within the *security group* as well as additional housekeeping operations are implemented through EOSIO WASM host functions, which can be invoked from wrapper actions within custom smart contracts.
```
Comment (bobgt) [inline docs/01_nodeos/07_concepts/15_private_chain_access/index.md:30]: Tiny typo: 
"Adding and removing participans" -> participants

File: docs/01_nodeos/07_concepts/15_private_chain_access/index.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+[[info | Technical note]]
+| Removing a participant from the *security group* means that a peer belonging to that participant will not get any data from the *security group* despite having a valid certificate. However, it will still be able to receive p2p handshake data such as head number, lib number, and other non-essential data.
```
Comment (bobgt) [inline docs/01_nodeos/07_concepts/15_private_chain_access/index.md:33]: p2p -> P2P or probably "...peer-to-peer(P2P)..."

File: docs/01_nodeos/07_concepts/15_private_chain_access/index.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+* [`add_security_group_participants()`](https://developers.eos.io/manuals/eosio.cdt/latest/namespaceeosio_1_1internal__use__do__not__use#function-add_security_group_participants)
+* [`remove_security_group_participants()`](https://developers.eos.io/manuals/eosio.cdt/latest/namespaceeosio_1_1internal__use__do__not__use#function-remove_security_group_participants)
+* [`in_active_security_group()`](https://developers.eos.io/manuals/eosio.cdt/latest/namespaceeosio_1_1internal__use__do__not__use#function-in_active_security_group)
+* [`get_active_security_group()`](https://developers.eos.io/manuals/eosio.cdt/latest/namespaceeosio_1_1internal__use__do__not__use#function-get_active_security_group)
+
+Note that adding or removing participants within the *security group* requires a privileged account. Also, note that these two actions are "proposed" for approval via EOSIO consensus. Finally, checking for existence of participants in the *security group* is all or nothing: if there is at least one specified participant that is not in the *security group*, the check returns false.
```
Comment (bobgt) [inline docs/01_nodeos/07_concepts/15_private_chain_access/index.md:48]: is false a value? Maybe can backtick it - "....the check returns `false`"

File: docs/01_nodeos/07_concepts/15_private_chain_access/index.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+Below are some of the potential use cases for *Private Access Control*, along with applicable business privacy requirements:
+
+* Consortium Solutions:
+  - Control on-chain user access and control via smart contracts.
+  - Allow storage and transmission of encrypted data among consortium members.
+  - Maintain a synced registry of approved vendors with all participants of the chain, with transparency on when a record is added or edited.
+  - Transaction recording and tracking using hash references of private chain data to a parent chain of consortium members.
+
+* Supply Chain Solutions:
+  - Private messaging between Clients to exchange confidential messages between trading partners by leveraging EPCIS technology and standards.
```
Comment (bobgt) [inline docs/01_nodeos/07_concepts/15_private_chain_access/index.md:70]: How about hyperlinking EPCIS?
https://www.gs1.org/standards/epcis

File: docs/01_nodeos/07_concepts/15_private_chain_access/index.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+  - Blockchain as a shared, immutable ledger to register the proof of the authenticity of transactions and execute smart contracts. The blockchain will enforce business rules, such as only one company can have legal ownership of a serialized unit at a given time.
+  - zk-SNARKs or another method of enhancing privacy to ensure no business data is revealed.
```
Comment (bobgt) [inline docs/01_nodeos/07_concepts/15_private_chain_access/index.md:72]: I think we can spell out the acronym and add the acronym in brackets:

Zero-Knowledge Succinct Non-Interactive Argument of Knowledge (zk-SNARKs)

That's a mouthful! :-) 

Or just hyperlink the acronym to an authoritative external page.

**Review by bobgt — APPROVED** [review]
Looks great! Added a few comments.

File: docs/01_nodeos/07_concepts/15_private_chain_access/index.md (hunk 2 above)
Comment (lparisc) [reply docs/01_nodeos/07_concepts/15_private_chain_access/index.md:8]: Should be "Private Access Control", which is mainly applicable to private chains. Will update title.

File: docs/01_nodeos/07_concepts/15_private_chain_access/index.md (hunk 4 above)
Comment (lparisc) [reply docs/01_nodeos/07_concepts/15_private_chain_access/index.md:30]: Oh man... :)

**Pull request merged**